components/Alert/AlertsAPI.tsx

```diff
@@ -0,0 +1,234 @@
+      },
+    }
+  }
+  const [alerts, setAlerts] = useState(props.alerts)
+  const [refMap] = useState(() => new WeakMap())
+  const [cancelMap] = useState(() => new WeakMap())
+
+  useEffect(() => {
+    setAlerts(props.alerts)
+  }, [props.alerts])
+
+  const transitionConfigs: any = {
+    from: { height: 0, opacity: 0, life: "100%" },
+    enter: (item: AlertProps) => async (next: any) =>
+      await next({ height: refMap.get(item).offsetHeight, opacity: 1 }),
+    leave: (item: AlertProps) => async (next: any, cancel: any) => {
+      cancelMap.set(item, cancel)
+      await next({ life: "0%" })
+      await next({ opacity: 0 })
+      await next({ height: 0 })
+    },
+    config: config,
+  }
+
+  const transitionProps = useTransition(
+    alerts,
+    item => item.key as string,
+    transitionConfigs
+  )
+
+  // Return container based on placement prop
+  const getWrapper = () => {
+    switch (props.placement) {
+      case "topLeft":
+        return TopLeftBox
+      case "topRight":
+        return TopRightBox
+      case "bottomLeft":
+        return BottomLeftBox
+      case "bottomRight":
+        return BottomRightBox
+      default:
+        return BottomLeftBox
+    }
+  }
+
+  const Wrapper = getWrapper()
+
+  return (
+    <Wrapper>
+      {transitionProps.map(({ key, item, props: { life, ...style } }: any) => {
+        let onClose = item.onClose
+        const curOnClose = onClose
+        onClose = (e?: Event): void => {
+          if (curOnClose) {
+            curOnClose(e)
+          }
+          if (cancelMap.has(item)) {
+            cancelMap.get(item)()
+          }
+        }
+        item.onClose = onClose
+
+        return (
+          <AnimatedDiv key={key} style={style}>
+            <Div ref={ref => ref && refMap.set(item, ref)}>
+              <Alert {...item} />
+            </Div>
+          </AnimatedDiv>
+        )
+      })}
+    </Wrapper>
+  )
+}
+
+// Wrapper Component for particular corner, Using React Class Component to access methods using API (callback ref).
+class AlertsWrapper extends React.Component<
+  AlertsWrapperProps,
+  AlertsWrapperStates
+> {
+  placement: placementType
+
+  constructor(props: AlertsWrapperProps) {
+    super(props)
+    this.state = {
+      alerts: [],
+    }
+    this.placement = this.props.placement ? this.props.placement : "bottomLeft"
+    this.add.bind(this)
+    this.remove.bind(this)
+    this.add.bind(this)
+  }
+
+  // Passed to API, add Alert to state and generate unique key
+  add = (alertProps: AlertProps): void => {
+    const key: string = generateKey()
+    alertProps = { key: key, ...alertProps }
+
+    // Closing Alert will call remove method of this class to update state
+    const curOnClose = alertProps.onClose
+    alertProps.onClose = event => {
+      this.remove(key)
+      if (curOnClose) {
+        curOnClose(event)
+      }
+    }
+
+    const newState = [alertProps, ...this.state.alerts]
+
+    this.setState({
+      alerts: newState,
```

Comment:
The dummy variable `newState` can be removed because it is implied that when you are calling `setState` you would be providing the new state.

```suggestion
      alerts: [alertProps, ...this.state.alerts],
```